pkg/solana/logpoller/log_poller.go

```diff
@@ -510,3 +513,32 @@ func (lp *Service) backgroundWorkerRun(ctx context.Context) {
 func (lp *Service) FilteredLogs(ctx context.Context, queryFilter []query.Expression, limitAndSort query.LimitAndSort, queryName string) ([]Log, error) {
 	return lp.orm.FilteredLogs(ctx, queryFilter, limitAndSort, queryName)
 }
+
+func (lp *Service) computeLookbackWindow(ctx context.Context) (int64, error) {
+	latestFinalizedSlot, err := lp.client.SlotHeightWithCommitment(ctx, rpc.CommitmentFinalized)
+	if err != nil {
+		return 0, fmt.Errorf("error getting latest slot from RPC: %w", err)
+	}
+
+	if latestFinalizedSlot == 0 {
+		return 0, fmt.Errorf("latest finalized slot is 0 - waiting for next slot to start processing")
+	}
+
+	lookback := lp.filters.MaxRetention()
+	if lookback == 0 { // Use default lookback if all filters have permanent retention or less than
+		lookback = DefaultLookbackWindow
+	}
+
+	firstAvailableSlot, err := lp.client.GetFirstAvailableBlock(ctx) // Despite the name, this returns slot # not block number
+
+	// nolint:gosec
+	// G115: integer overflow conversion uint64 -&gt; int64
+	latestSlot, firstSlot := int64(latestFinalizedSlot), int64(firstAvailableSlot)
+	lookbackSlot := latestSlot - int64((lookback-1)/lp.blockTime) - 1 // = latestSlot - ceil(lookback/blockTime)
```

Comment:
Arent division for integers already ceil? `int(10/3) = 3`

Reply:
They're floor. We want `ceil(10/3) = 4`

Reply:
> You probably meant int64((lookback-1)/lp.blockTime) + 1

No, the reason I went with ceil instead of floor is that it seems better to err on the side of including a block rather than excluding it if `lookback/blockTime` doesn't come out to an even number. Using +1 here 

> But int64((lookback-1)/lp.blockTime) - 1, where lookback = 10 and blockTime = 3 results in 2 .

That's different from what's in the code.  I think you're forgetting about the - sign in front of the int64. It's:
```
latestSlot -  int64(lookback-1)/lp.blockTime) - 1 
```
So for your example:  `latestSlot - 3 - 1 = latestSlot - 4`

Reply:
> You probably meant int64((lookback-1)/lp.blockTime) + 1

Yes, in that it's lookbackSlot minus this

Reply:
Yes, you are right. I've missed the - sign in front of the int64